python/integration_tests/test_task_worker_processing.py

```diff
@@ -0,0 +1,278 @@
+import pytest
+import signal
+import sqlite3
+import subprocess
+import threading
+import time
+
+import yaml
+
+from pathlib import Path
+from python.integration_tests.helpers import (
+    TASKBROKER_BIN,
+    send_messages_to_kafka,
+    check_topic_exists,
+    create_topic,
+    recreate_topic,
+)
+
+from python.integration_tests.worker import SimpleTaskWorker, TaskWorkerClient
+
+
+TEST_OUTPUT_PATH = Path(__file__).parent / ".output_from_test_task_worker_processing"
+processed_tasks = set()
+
+
+def manage_taskworker(
+    worker_id: int,
+    consumer_config: dict,
+    log_file_path: str,
+    tasks_written_event: threading.Event,
+    shutdown_event: threading.Event
+) -> None:
+    print(f"[taskworker_{worker_id}] Starting taskworker_{worker_id}")
+    worker = SimpleTaskWorker(TaskWorkerClient(f"127.0.0.1:{consumer_config['grpc_port']}"))
+    fetched_tasks = 0
+    completed_tasks = 0
+
+    next_task = None
+    task = None
+
+    # Wait for consumer to initialize sqlite and write tasks to it
+    print(f"[taskworker_{worker_id}]: Waiting for consumer to initialize sqlite and write tasks to it...")
+    while not tasks_written_event.is_set() and not shutdown_event.is_set():
+        time.sleep(1)
+
+    if not shutdown_event.is_set():
+        print(f"[taskworker_{worker_id}]: Processing tasks...")
+    try:
+        while not shutdown_event.is_set():
+            if next_task:
+                task = next_task
+                next_task = None
+                fetched_tasks += 1
+            else:
+                task = worker.fetch_task()
+                if task:
+                    fetched_tasks += 1
+            if not task:
+                if get_num_tasks_by_status(consumer_config, 'Pending') == 0:
```

Comment:
Do we need to do this check? Once a worker has gotten an empty response, we can infer that there are no pending tasks remaining and can shutdown the worker. This could be contributing to contention in higher worker counts.

Reply:
You're right, we don't need this check anymore. Originally, I had the workers processing messages in parallel with the consumer inserting into the inflight store. Now that we wait for all messages to be written first before processing, we don't need this check.